tools/github_update.py

```diff
@@ -198,12 +199,16 @@ def list_contexts(self):
         '''returns a set of context labels of all statuses in a given commit'''
         if not self._api:
             return set([])
-        return set([status['context'] for status in self._api.get_commit_statuses()])
+        try:
+            return set([status['context'] for status in self._api.get_commit_statuses()])
+        except:
+            # in the event of network error, just log the exception and continue
+            traceback.print_exc()
```

Comment:
I'm slightly obsessive about format_exc() and reporting it via standard channels.  Take or leave.